src/plugins/event-plugins/XhrPlugin.ts

```diff
@@ -203,25 +212,31 @@ export class XhrPlugin extends MonkeyPatched<XMLHttpRequest, 'send' | 'open'> {
     private handleXhrAbortEvent = (e: Event) => {
         const xhr: XMLHttpRequest = e.target as XMLHttpRequest;
         const xhrDetails = this.xhrMap.get(xhr);
-        const errorName = 'XMLHttpRequest abort';
-        this.handleXhrDetailsOnError(xhrDetails, errorName);
+        if (xhrDetails) {
+            this.fillEndTimeIfEmpty(xhrDetails);
+            const errorName = 'XMLHttpRequest abort';
+            this.handleXhrDetailsOnError(xhrDetails, errorName);
+        }
     };
 
     private handleXhrTimeoutEvent = (e: Event) => {
         const xhr: XMLHttpRequest = e.target as XMLHttpRequest;
         const xhrDetails = this.xhrMap.get(xhr);
-        const errorName = 'XMLHttpRequest timeout';
-        this.handleXhrDetailsOnError(xhrDetails, errorName);
+        if (xhrDetails) {
+            this.fillEndTimeIfEmpty(xhrDetails);
```

Comment:
question: Do we need this refactoring? I see that we have the same check at line 236-237. Is this to get the most accurate end time?

Reply:
yes. date.now() must be called as early as possible